src/encoded/types/biosample_type.py

```diff
@@ -0,0 +1,112 @@
+from snovault import (
+    calculated_property,
+    collection,
+    load_schema,
+)
+from .base import (
+    SharedItem,
+)
+
+
+@collection(
+    name='biosample-types',
+    unique_key='biosample_type:name',
+    properties={
+        'title': 'Biosample type',
+        'description': 'General types of biosample used in the ENCODE project',
+    })
+class BiosampleType(SharedItem):
+    item_type = 'biosample_type'
+    schema = load_schema('encoded:schemas/biosample_type.json')
+    embedded = ['references']
+
+    def unique_keys(self, properties):
+        keys = super(BiosampleType, self).unique_keys(properties)
+        keys.setdefault('biosample_type:name', []).append(self.name(properties))
+        return keys
+
+    @calculated_property(schema={
+        "title": "Name",
+        "type": "string",
+    })
+    def name(self, properties=None):
+        if properties is None:
+            properties = self.upgrade_properties()
+        return u'{}_{}'.format(
+            properties['classification'],
+            '_'.join(properties['term_ids'])
+        ).replace(' ', '_').replace(':', '_')
+
+    @property
+    def __name__(self):
+        return self.name()
+
+    @calculated_property(condition='term_ids', schema={
+        "title": "Organ slims",
+        "type": "array",
+        "items": {
+            "type": "string",
+        },
+    })
+    def organ_slims(self, registry, term_ids):
```

Comment:
Did you consider putting this in a static function?

For example,
```
@staticmethod
def _get_ontology_slims(registry, term_ids, slim_key):
    slims = set()
    for term_id in term_ids:
        if term_id in registry['ontology']:
            slims.update(registry['ontology'][term_id][slim_key])
    return list(slims)

@calculated_property(condition='term_ids', schema={
    ...
})
def organ_slims(self, registry, term_ids):
    return self._get_ontology_slims(registry, term_ids, 'organs')

@calculated_property(condition='term_ids', schema={
    ...
})
def cell_slims(self, registry, term_ids):
    return self._get_ontology_slims(registry, term_ids, 'cells')

etc...
```
We're doing the same pattern 5 times with just a change of key value.

Then try set comprehension too
```
@staticmethod
def _get_ontology_slims(registry, term_ids, slim_key):
    slims = set(
        registry['ontology'][term_id][slim_key]
        for term_id in term_ids
        if term_id in registry['ontology']
    )
    return list(slims) 
```